setup_monitoring.sh

```diff
@@ -0,0 +1,103 @@
+  installPlanApproval: Automatic
+  name: rhods-prometheus-operator
+  source: redhat-operators
+  sourceNamespace: openshift-marketplace
+  startingCSV: rhods-prometheus-operator.4.10.0
+EOF
+# create a subsctiption for prometheus-exporter operator
+oc apply -f - <<EOF
+---
+apiVersion: operators.coreos.com/v1alpha1
+kind: Subscription
+metadata:
+  name: prometheus-exporter-operator
+  namespace: 3scale-test
+spec:
+  channel: alpha
+  installPlanApproval: Automatic
+  name: prometheus-exporter-operator
+  source: community-operators
+  sourceNamespace: openshift-marketplace
+  startingCSV: prometheus-exporter-operator.v0.7.0
+EOF
+# Create Grafana Subscription
+oc apply -f - <<EOF
+---
+apiVersion: operators.coreos.com/v1alpha1
+kind: Subscription
+metadata:
+  name: grafana-operator
+  namespace: 3scale-test
+spec:
+  channel: v4
+  installPlanApproval: Automatic
+  name: grafana-operator
+  source: community-operators
+  sourceNamespace: openshift-marketplace
+  startingCSV: grafana-operator.v4.10.1
+EOF
+# Get the SECRET name that contains the THANOS_QUERIER_BEARER_TOKEN
+SECRET=`oc get secret -n openshift-user-workload-monitoring | grep  prometheus-user-workload-token | head -n 1 | awk '{print $1 }'`
+# Get the THANOS_QUERIER_BEARER_TOKEN using the SECRET name
+THANOS_QUERIER_BEARER_TOKEN=$(oc get secret $SECRET -n openshift-user-workload-monitoring -o jsonpath="{.data.token}" | base64 -d)
+# patch the THANOS_QUERIER_BEARER_TOKEN in the 3scale-scrape-configs.yaml
+sed -i "s|<THANOS_QUERIER_BEARER_TOKEN>|$THANOS_QUERIER_BEARER_TOKEN|g" 3scale-scrape-configs.yaml
+# create secret addition-scrape-configs from 3scale-scrape-configs.yaml file
+oc create secret generic additional-scrape-configs --from-file=3scale-scrape-configs.yaml=./3scale-scrape-configs.yaml
+
+# prometheus exporter 
+BKEND_REDIS_URL=$(oc get secret backend-redis -n 3scale-test -o jsonpath={.data.REDIS_QUEUES_URL} | base64 -d)
+BACKEND_REDIS_URL=$(echo "$BKEND_REDIS_URL" | sed 's|^redis://||; s|:6379$||')
+
+oc apply -n 3scale-test -f - <<EOF   
+---
+apiVersion: monitoring.3scale.net/v1alpha1
+kind: PrometheusExporter
+metadata:
+  name: backend-redis
+spec:
+  type: redis
+  grafanaDashboard:
+    label:
+      key: monitoring-key
+      value: middleware
+  extraLabel:
+    key: threescale_component
+    value: backend
+  dbHost: $BACKEND_REDIS_URL
+  dbPort: 6379
+  dbCheckKeys: "db1=resque:queue:stats,db1=resque:queue:priority,db1=resque:queue:main,db1=resque:failed,db1=resque:workers,db1=resque:queues"
+EOF
+# Prometheus CR
+DOMAIN=$(oc get routes console -n openshift-console -o json | jq -r '.status.ingress[0].routerCanonicalHostname' | sed 's/router-default.//')
+EXTERNALURL=https://prometheus.3scale-test.$DOMAIN
+sed -i "s|externalUrl:.*|externalUrl: $EXTERNALURL|" prometheus.yaml
+oc apply -f prometheus.yaml
+oc expose service prometheus-operated --hostname prometheus.3scale-test.$DOMAIN
+# Grafana CR's
+oc apply -f datasource.yaml
+oc apply -f grafana.yaml
```

Comment:
It doesn't look like the Grafana CRDs are being correctly loaded so these commands fail unless you manually apply the CRDs. Here's the output:
```
error: resource mapping not found for name: "prometheus" namespace: "" from "datasource.yaml": no matches for kind "GrafanaDataSource" in version "integreatly.org/v1alpha1"
ensure CRDs are installed first
error: resource mapping not found for name: "example-grafana" namespace: "" from "grafana.yaml": no matches for kind "Grafana" in version "integreatly.org/v1alpha1"
ensure CRDs are installed first
```

Reply:
hmm did see this when I ran it . May be a timing/race condition

Reply:
looks like the scope is on the operator is defaulting to clusterscoped and this is causing the operator install to fail :thinking: guess I need to patch the operatorGroup to stop this .

Reply:
weird issue where only a cluster scoped 3scale-operator operatorGroup existed and that was effecting the scope on other subscriptions for operators when installing. Need to set the scope on the 3scale operatorGroup

Reply:
Ok think I have figured it out the operatorGroups annotations get updated for every subscription and so this one operatorGroup covers all subscription. 
```yaml
annotations:
    olm.providedAPIs: 'APIManager.v1alpha1.apps.3scale.net,APIManagerBackup.v1alpha1.apps.3scale.net,APIManagerRestore.v1alpha1.apps.3scale.net,ActiveDoc.v1beta1.capabilities.3scale.net,Application.v1beta1.capabilities.3scale.net,Backend.v1beta1.capabilities.3scale.net,CustomPolicyDefinition.v1beta1.capabilities.3scale.net,DeveloperAccount.v1beta1.capabilities.3scale.net,DeveloperUser.v1beta1.capabilities.3scale.net,Grafana.v1alpha1.integreatly.org,GrafanaDashboard.v1alpha1.integreatly.org,GrafanaDataSource.v1alpha1.integreatly.org,GrafanaFolder.v1alpha1.integreatly.org,GrafanaNotificationChannel.v1alpha1.integreatly.org,OpenAPI.v1beta1.capabilities.3scale.net,Product.v1beta1.capabilities.3scale.net,PrometheusExporter.v1alpha1.monitoring.3scale.net,ProxyConfigPromote.v1beta1.capabilities.3scale.net,Tenant.v1alpha1.capabilities.3scale.net'
```
I taught every fresh subscription created a new operatorGroup which would make more sense. :thinking:

Reply:
anyway the answer is the same patch the operatorGroup to namespaced scope.

Reply:
@carlkyrillos we won't be load testing against 2.14 so will changed the deployment of 3scale-operator to a dev image built of master.